language-design.md

```diff
@@ -38,17 +64,28 @@ How do the themes of _Growing a Language_ relate to the lab we did this week?
+
+Another interesting similarity is that Steele defined words in two ways: either by directly defining words, or by defining ways in which words can build into other words (such as adding -ing to verbs). Within the lab, we could do something similar with our design: we could add more building blocks (things similar to changing the volume or reversing), or we could add new ways to combine those building blocks. By doing both of these in conjunction, we can make our language as easy as possible for users to grow.
+
 
 
 ---
  
 **Question**
 
-
 In what way is an API a language? 
 
 **Response**
 
+Like a language, an API is at its core a vocabulary of words you can pass into the API, and meanings for each of those words. In both cases, what’s important is how these words are interpreted in the context of the language and how the user understands the corresponding documentation. An API is essentially a “little language.” Writing a call to an API feels like writing a call to a method in any standard language. This is much like one of the criteria we discussed for DSLs: they have to have a language feel to them [Fowler 2010]. The fact that APIs have a language feel to them is good evidence in favor of their being considered as languages on the same level as DSLs.
+
+Just like any language, an API is growable in the way that Steele discussed, namely in his example of one syllable words and additional definitions. Like our third quote in the first question mentions, you can always add additional functionality to an API as more of a need for it becomes apparent. In fact, it may even be easier with an API because there are fewer interactions between user and language to consider. 
+
+Another big similarity is that language design and API design are very similar processes. Like Bloch says, there are many many standard practices in the world of API design that are similar to good practices for language design. Both of them need to be very focused on what the user really needs, rather than what the language designer thinks that the user needs. Both should be as simple and elegant as possible while still having all of the functionality that they need. Likewise, languages and APIs must be properly documented in a way that makes sense to the user. If the user cannot understand the documentation, they are less likely to use the language or realize all its capabilities. 
+
 
```

Comment:
You guys bring up some really strong similarities between API's and languages. I think it could also be interesting to bring up what the differences are and to explain how these compare to the similarities. (Personally, I think there are more similarities than differences and I'd be curious to see what you think.)